binderhub/builder.py

```diff
@@ -228,6 +228,24 @@ def set_default_headers(self):
         self.set_header("content-type", "text/event-stream")
         self.set_header("cache-control", "no-cache")
 
+    def _get_build_only_outcome(self):
```

Comment:
I'd call this `_get_build_only()`, since it's the dynamic resolution of the value of `build_only` with checks, etc., not a separate concept called 'outcome', which confused me a bit.